src/opm/parser/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -331,18 +521,436 @@ namespace Opm {
+            coord.push_back(zt);
+            coord.push_back(x0);
+            coord.push_back(yb);
+            coord.push_back(zb);
+
+            for (int i=0; i < dims[0]; i++) {
+
+                ind = i+(j+1)*dims[0]+1;
+
+                if (j == (dims[1]-1) ) {
+                    ind = i+j*dims[0]+1;
+                }
+
+                if (i == (dims[0] - 1) ) {
+                    ind=ind-1;
+                }
+
+                zt = TOPS[ind];
+                zb = zt + sumDZ(i, j, dims, DZ);
+
+                double xt=-999;
+                double xb=-999;
+
+                if (j == (dims[1]-1) ) {
+                    xt = sumDX(j, 0, i, dims, DX);
+                    xb = sumDX(j, dims[2]-1, i, dims, DX);
+                } else {
+                    xt = sumDX(j+1, 0, i, dims, DX);
+                    xb = sumDX(j+1, dims[2]-1, i, dims, DX);
+                }
+
+                if (i == (dims[0] - 1) ) {
+                    yt = sumDY(i, 0, j, dims, DY);
+                    yb = sumDY(i, dims[2]-1, j, dims, DY);
+                } else {
+                    yt = sumDY(i+1, 0, j, dims, DY);
+                    yb = sumDY(i+1, dims[2]-1, j, dims, DY);
+                }
+
+                coord.push_back(xt);
+                coord.push_back(yt);
+                coord.push_back(zt);
+                coord.push_back(xb);
+                coord.push_back(yb);
+                coord.push_back(zb);
+
+                x0 = xt;
+            }
+        }
+
+        return coord;
+    };
+
+    std::vector<double> EclipseGrid::makeZcornDzTops(const std::array<int, 3>& dims, const std::vector<double>& DZ, const std::vector<double>& TOPS) {
+
+        std::vector<double> zcorn;
+        size_t sizeZcorn = dims[0]*dims[1]*dims[2]*8;
+
+        zcorn.assign (sizeZcorn, 0.0);
+
+        for (int j = 0; j < dims[1]; j++) {
+            for (int i = 0; i < dims[0]; i++) {
+                int ind = i + j*dims[0];
+                double z = TOPS[ind];
+
+                for (int k = 0; k < dims[2]; k++) {
+
+                    // top face of cell
+                    int zind = i*2 + j*dims[0]*4 + k*dims[0]*dims[1]*8;
+
+                    zcorn[zind] = z;
+                    zcorn[zind + 1] = z;
+
+                    zind = zind + dims[0]*2;
+
+                    zcorn[zind] = z;
+                    zcorn[zind + 1] = z;
+
+                    z = z + DZ[i + j*dims[0] + k*dims[0]*dims[1]];
+
+                    // bottom face of cell
+                    zind = i*2 + j*dims[0]*4 + k*dims[0]*dims[1]*8 + dims[0]*dims[1]*4;
+
+                    zcorn[zind] = z;
+                    zcorn[zind + 1] = z;
+
+                    zind = zind + dims[0]*2;
+
+                    zcorn[zind] = z;
+                    zcorn[zind + 1] = z;
+                }
+            }
+        }
+
+        return zcorn;
+    };
+
+    double EclipseGrid::sumDX(int j, int k, int i1, const std::array<int, 3>& dims, const std::vector<double>& dx) {
```

Comment:
The order of the arguments here is prone to errors, but it is probably correctly used.

Reply:
It is so error prone that it should be changed - in my opinion.

Reply:
fixed both for sumDX and sumDY.